Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -114,41 +117,136 @@ namespace PhysX
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySets)
+    {
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        // Get the original velocity so we can determine that set actually changes things.
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        // Set the falling velocity to the expected + original velocity to verify that the velociy will change.
+        basis.m_gameplayController->SetFallingVelocity(originalVelocity + expectedVelocity);
+
+        // Get the end velocity and validate that it is the expected value.
+        auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+        EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(expectedVelocity.GetZ(), 0.001f));
+    }
+
+    //! Test the verifies that the Falling Velocity sets and uses the set value to interact with the tick incrementing.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySetsWhileMoving)
+    {
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        // Get the original velocity so we can determine that set actually changes things.
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        // Run the scene for a while and validate that the set velocity is working
+        for (int i = 0; i < 10; i++)
+        {
+            // Calculate and set velocity based upon the original velocity, expected velocity, and loop count modifier
+            auto setVelocity = originalVelocity + expectedVelocity + AZ::Vector3(0.0f, 0.0f, float(i));
+            basis.m_gameplayController->SetFallingVelocity(setVelocity);
+
+            basis.Update();
+
+            // Get the endVelocity & current gravity to perform calculations on.
+            auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+            const auto gravity = basis.m_testScene->GetGravity();
+
+            // The end velocity should be decelerated by gt when g is gravity acceleration and t is the length of time
+            const float expectedDeceleration = gravity.GetZ() * basis.m_timeStep;
+
+            // Verify that the end velocity is the set velocity minus the deceleration for a single time step
+            EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(setVelocity.GetZ() + expectedDeceleration, 0.001f));
+        }
+    }
+
+    //! Test that verifies that the Ground Detec Set & Get methods function as expected.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_SetGroundDetectionHeight)
+    {
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        const float expectedHeight = 10.0f;
```

Comment:
Nitpick - doesn't really have any impact on this test, but this is an unrealistically large value for the ground detection box height (it's much taller than the character).

Reply:
I was intentionally using a large number to make it obvious that the value was set.